src/TimingMasterController.cpp

```diff
@@ -52,23 +55,25 @@ TimingMasterController::TimingMasterController(const std::string& name)
 void
 TimingMasterController::do_configure(const nlohmann::json& data)
 {
-  auto conf = data.get<timingmastercontroller::ConfParams>();
-  if (conf.device.empty())
+  // auto conf = data.get<timingmastercontroller::ConfParams>();
+  auto mdal = m_params->module<dal::TimingMasterController>(get_name()); 
+
+  if (mdal->get_device_str().empty())
   {
     throw UHALDeviceNameIssue(ERS_HERE, "Device name should not be empty");
   }
-  m_timing_device = conf.device;
-  m_hardware_state_recovery_enabled = conf.hardware_state_recovery_enabled;
-  m_timing_session_name = conf.timing_session_name;
-  m_monitored_endpoint_locations = conf.monitored_endpoints;
+  m_timing_device = mdal->get_device_str();
+  m_hardware_state_recovery_enabled = mdal->get_hardware_state_recovery_enabled();
+  m_timing_session_name = mdal->get_timing_session_name();
+  // m_monitored_endpoint_locations = mdal->get_monitored_endpoints();
```

Comment:
for this, I need to change the type I've used in the schema file to be of a vector format but I am unsure how to do this and what the available types are... I've looked through this: https://gitlab.cern.ch/atlas-tdaq-software/oks/-/blob/master/doc/RELEASE_NOTES.md but it seems it's just strings/ints - could I make a `std::vector<std::s16> ` for example?

Reply:
If you are editing the schema in a text editor you just need to set `id-multi-value="yes"` in the attribute definition. If you are using dbe then you just need to toggle the `Is Multivariable` flag below the Name and Type fields. The generated code will give you a getter that returns a vector rather than a single value.

Reply:
thank you! I've changed the definition to: ` <attribute name="monitored_endpoints" description="List of monitored endpoint addresses" type="string" is-multi-value="yes" init-value="Unknown"/>`  but it doesn't look like the dal file has it updated as the definition is still: 
        private:

          uint32_t m_endpoint_scan_period;
          std::string m_clock_config;
          bool m_soft;
          std::string m_fanout_mode;
          std::string m_monitored_endpoints;
 what am I doing wrong if just running dbt-build isn't resetting the values?

Reply:
Try running `dbt-build -c`. Sometimes it seems to not update everything it should.

Reply:
thank you, this did help load in a bunch of new changes I'd made to the schema file!